code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityExtension.java

```diff
@@ -13,232 +13,206 @@
-				if (EventUtils.isGetUrlVariablesRequestEvent(event)) {
-					handleUrlVariablesRequest(event);
-				} else {
-					handleIdentityRequest(event);
-				}
-			} else if (EventUtils.isRequestContentEvent(event)) {
-				handleRequestContent(event);
-			} else if (EventUtils.isUpdateIdentityEvent(event)) {
-				handleUpdateIdentities(event);
-			} else if (EventUtils.isRemoveIdentityEvent(event)) {
-				handleRemoveIdentity(event);
-			} else if (EventUtils.isRequestResetEvent(event)) {
-				handleRequestReset(event);
-			} else if (EventUtils.isSharedStateUpdateFor(IdentityConstants.SharedState.IdentityDirect.NAME, event)) {
-				handleIdentityDirectECIDUpdate(event);
-			}
+		getApi()
+			.registerEventListener(
+				IdentityConstants.EventType.EDGE_IDENTITY,
+				IdentityConstants.EventSource.REMOVE_IDENTITY,
+				this::handleEvent
+			);
 
-			cachedEvents.poll();
-		}
+		// HUB shared state event listener
+		getApi()
+			.registerEventListener(
+				IdentityConstants.EventType.HUB,
+				IdentityConstants.EventSource.SHARED_STATE,
+				this::handleEvent
+			);
+	}
+
+	@Override
+	public boolean readyForEvent(@NonNull Event event) {
+		// Check if we are already booted
+		if (state.hasBooted()) return true;
+
+		// Attempt to boot
+		return state.bootupIfReady(sharedStateHandle);
 	}
 
 	/**
-	 * Calls {@link IdentityState#bootupIfReady(SharedStateCallback)} with a valid callback
-	 *
-	 * @return True if the bootup is complete
+	 * Responsible for handling the incoming events that this extension is registered for.
 	 */
-	boolean bootupIfReady() {
-		final SharedStateCallback callback = createSharedStateCallback();
-
-		return state.bootupIfReady(callback);
+	@VisibleForTesting
+	void handleEvent(@NonNull final Event event) {
+		if (EventUtils.isRequestIdentityEvent(event)) {
+			if (EventUtils.isGetUrlVariablesRequestEvent(event)) {
+				handleUrlVariablesRequest(event);
+			} else {
+				handleIdentityRequest(event);
+			}
+		} else if (EventUtils.isRequestContentEvent(event)) {
+			handleRequestContent(event);
+		} else if (EventUtils.isUpdateIdentityEvent(event)) {
+			handleUpdateIdentities(event);
+		} else if (EventUtils.isRemoveIdentityEvent(event)) {
+			handleRemoveIdentity(event);
+		} else if (EventUtils.isRequestResetEvent(event)) {
+			handleRequestReset(event);
+		} else if (EventUtils.isSharedStateUpdateFor(IdentityConstants.SharedState.IdentityDirect.NAME, event)) {
+			handleIdentityDirectECIDUpdate(event);
+		}
 	}
 
 	/**
 	 * Handles events requesting for formatted and encoded identifiers url for hybrid apps.
 	 *
 	 * @param event the identity request {@link Event}
 	 */
-	void handleUrlVariablesRequest(final Event event) {
-		String urlVariablesString = null;
-
-		final Map<String, Object> configurationState = getSharedState(
+	private void handleUrlVariablesRequest(final Event event) {
+		final SharedStateResult configSharedStateResult = sharedStateHandle.getSharedState(
```

Comment:
this type of event requires config shared state, can we add a check for its availability in readyForEvent?